sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/graph/UserPrincipalMicrosoftGraphTests.java

```diff
@@ -84,8 +85,8 @@ void setup() {
     @Test
     void getGroups() throws Exception {
         properties.getUserGroup().setAllowedGroupNames(Arrays.asList("group1", "group2", "group3"));
-        AadGraphClient graphClientMock = new AadGraphClient(clientId, clientSecret, properties,
-            endpoints);
+        AadGraphClient graphClientMock = new AadGraphClient(
+                clientId, clientSecret, properties, endpoints, new RestTemplate());
```

Comment:
Does this mean we don't need wiremock, we can directly mock the RestTemplate here?

Reply:
With wiremock, the http request can be validated. We need to validate the http request.

Here is the screenshot of **using wiremock** vs **mock TestTemplate** only:

![image](https://user-images.githubusercontent.com/13167207/191403101-4613f6a5-9315-4500-b771-f41633888d13.png)

Reply:
Would something like this https://www.baeldung.com/spring-mock-rest-template#using-spring-test work?

Reply:
Will do.